src/main/webapp/webapptests/spec/SpecAvailability.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe("Availability", function() {
+
+    describe("when the user requests to see a tutor's availability", function() {
+        var mockWindow = {location: {href: "availability.html?tutorID=test%40gmail.com", search: "?tutorID=test%40gmail.com"}};
+
+        it("should trigger the fetch function", function() {
+            spyOn(window, "onload").and.callFake(function() {
+                readTutorID(queryString, window);
+            });
+            spyOn(window, 'fetch').and.callThrough();
```

Comment:
I read that callThrough() executes the real implementation. Should we be doing that in tests?

Reply:
In this case, I wanted fetch to run normally. The reason I created the spy was to be able to know in the end if had been successfully executed. I wouldn't be able to do expect(window.fetch).toHaveBeenCalled() if I hadn't spied on fetch.

Reply:
Ok, that makes sense!